packages/runtime/container-runtime/src/summary/summaryManager.ts

```diff
@@ -238,6 +238,10 @@ export class SummaryManager extends TypedEventEmitter<ISummarizerEvents> impleme
 
 		this.delayBeforeCreatingSummarizer()
 			.then(async (startWithInitialDelay: boolean) => {
+				if (this.disposed) {
```

Comment:
This is redundant with what's happening inside `getShouldSummarizeState`. I would keep either or. You can update the `stopReason` in that funciton to return `disposed`.

Reply:
It's not.  That's called in many places and needs to protect against disposed.  This needs to ensure it doesn't run if disposed (look closely at usages of line 252's variable)

Reply:
i think i agree with @agarwal-navin, and we only need this in getShouldSummarizeState. all the code here is synchronous, so exiting earlier doesn't do much

Reply:
If `startWithInitialDelay` is false it won't exit early without this.